### PR TITLE
Improved code coverage

### PR DESCRIPTION
**Pull request by albi3ro**
This PR handles several lines that were missing code coverage by either:
* explicitly specifying the lines will not be covered by our tests
* fixing logic that won't ever be covered
* adding tests

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -45,6 +45,8 @@
 * `Operator.decomposition()` is now an instance method, and no longer accepts parameters.
   [(#2498)](https://github.com/PennyLaneAI/pennylane/pull/2498)
 
+* Adds tests, adds no-coverage directives, and removes inaccessible logic to improve code coverage.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:48]:
```suggestion
* Adds tests, adds no-coverage directives, and removes inaccessible logic to improve code coverage.
  [(#2537)](https://github.com/PennyLaneAI/pennylane/pull/2537)
```

File: pennylane/ops/qubit/hamiltonian.py [hunk 2]
```diff
@@ -621,8 +621,6 @@ def queue(self, context=qml.QueuingContext):
             except QueuingError:
                 o.queue(context=context)
                 context.update_info(o, owner=self)
-            except NotImplementedError:
```
Comment (albi3ro) [inline pennylane/ops/qubit/hamiltonian.py:624]: Hamiltonians contain observables. Observables will have a `queue` method.

File: tests/fourier/test_coefficients.py [hunk 3]
```diff
@@ -258,7 +258,7 @@ def test_anti_aliasing(self, circuit, inpt, degree):
         """Test that the coefficients obtained through anti-aliasing are the
         same as the ones when we don't anti-alias at the correct degree."""
         coeffs_regular = coefficients(circuit, len(inpt), degree, lowpass_filter=False)
-        coeffs_anti_aliased = coefficients(circuit, len(inpt), degree)
+        coeffs_anti_aliased = coefficients(circuit, len(inpt), degree, lowpass_filter=True)
```
Comment (antalszava) [inline tests/fourier/test_coefficients.py:261]: What does `lowpass_filter` do? :thinking:

File: tests/ops/functions/test_generator.py [hunk 4]
```diff
@@ -176,6 +176,18 @@ class DeprecatedClassOp(CustomOp):
         assert gen.name == "Hermitian"
         assert gen.wires.tolist() == ["a"]
 
+    def test_generator_property_old_default(self):
+        """Test that if the old-style generator property is the old default,
```
Comment (antalszava) [inline tests/ops/functions/test_generator.py:180]: (Perhaps intentional, if so, ignore this suggestion)
```suggestion
        """Test that if the old-style generator property is the default,
```

**Review by antalszava — APPROVED** [review]
Looks good! :tada: Thank you :slightly_smiling_face:

File: tests/fourier/test_coefficients.py (hunk 3 above)
Comment (albi3ro) [reply tests/fourier/test_coefficients.py:261]:
```
lowpass_filter (bool) – If True, a simple low-pass filter is applied prior to computing the set of coefficients in order to filter out frequencies above the given degree. See examples below.
```
It defaults to `False`.  So `coeffs_anti_aliased` were not in fact anti-aliased, but regular.

**Pull request merged**